src/pages/Profile/Profile.js

```diff
@@ -0,0 +1,155 @@
+
+  const [userData, setUserData] = useState({
+    displayName: user.displayName,
+    email: user.email,
+  })
+  const [confirmationPassword, setConfirmationPassword] = useState('')
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false)
+  const [errorMessage, setErrorMessage] = useState('')
+
+  async function updateUserProfileFirebase(userData) {
+    const user = firebase.auth().currentUser
+    try {
+      await user.updateProfile({
+        displayName: userData.displayName,
+      })
+      if (userData.email !== user.email) {
+        openModal()
+      }
+    } catch (error) {
+      setErrorMessage(`Updating ${user.displayName} failed. ${error}`)
+    }
+  }
+
+  async function updateEmailWithFirebase() {
+    if (!confirmationPassword) {
+      setErrorMessage('Password not confirmed!')
+      return
+    }
+
+    const credentials = firebase.auth.EmailAuthProvider.credential(
+      user.email,
+      confirmationPassword
+    )
+    try {
+      await user.reauthenticateWithCredential(credentials)
+      await user.updateEmail(userData.email)
+    } catch (error) {
+      console.log(error.message)
+      setErrorMessage(`Email Address could not be updated. ${error}`)
+    } finally {
+      closeModal()
+    }
+  }
+
+  function updateEmailForm() {
+    const user = firebase.auth().currentUser
+    setUserData({ ...userData, email: user.email })
+    setConfirmationPassword('')
+  }
+
+  async function logoutFromFirebase() {
+    try {
+      navigateTo(ROUTES.WELCOME)
+      await firebaseApp.signOut()
+    } catch (error) {
+      setErrorMessage('error.message')
+    }
+  }
+
+  return (
+    <StyledMain>
+      {showConfirmPassword && (
+        <ConfirmPasswordModal
+          confirmationPassword={confirmationPassword}
+          updateEmailWithFirebase={updateEmailWithFirebase}
+          updateConfirmationPassword={updateConfirmationPassword}
+          closeModal={closeModal}
+        />
+      )}
+      <Link to={ROUTES.ADD_TRIP}>
+        <span>Go back</span>
+      </Link>
+      <h2>User Profile</h2>
+      <ProfileIcon>
+        <img src={profileIcon} alt="profile" />
+      </ProfileIcon>
+
+      <EditProfileForm
+        updateUserProfileFirebase={updateUserProfileFirebase}
+        userData={userData}
+        setUserData={updateUserData}
+        errorMessage={errorMessage}
+      />
+      <Button onClick={logoutFromFirebase} text="Log out" />
+    </StyledMain>
+  )
+
+  function closeModal() {
+    updateEmailForm()
+    setShowConfirmPassword(false)
+  }
+  function openModal() {
+    setShowConfirmPassword(true)
+  }
+
+  function updateConfirmationPassword(input) {
+    setConfirmationPassword(input)
+  }
+
+  function updateUserData(input) {
+    setUserData(input)
+  }
+}
+
+const StyledMain = styled.main`
+  position: relative;
+  background: var(--sand);
+  height: 100vh;
+  padding: 30px;
+  display: flex;
+  flex-direction: column;
+
+  h2 {
```

Comment:
Ich hab jetzt dein Styling nicht im Detail angeguckt, denn in der App selbst sah für mich alles sehr rund aus :) Hier ist mir nur gerade aufgefallen, dass du die h-Elemente auch in den Global Styles definieren könntest. Also zumindest die font und die Größe zum Beispiel.

`h1, h2, h3, h4, h5: {
font-family: var(--headlineFont);
}
`